src/KimeraVioRos.cpp

```diff
@@ -157,41 +169,15 @@ KimeraVioRos::createDataProvider(const VioParams& vio_params) {
   return nullptr;
 }
 
-void KimeraVioRos::connectVioPipelineAndDataProvider() {
-  CHECK(data_provider_);
-  CHECK(vio_pipeline_);
-
+void KimeraVioRos::connectVIO() {
   // Register VIO pipeline callbacks
   // Register callback to shutdown data provider in case VIO pipeline
   // shutsdown.
+  CHECK(data_provider_);
+  CHECK(vio_pipeline_);
   vio_pipeline_->registerShutdownCallback(std::bind(
       &VIO::DataProviderInterface::shutdown, std::ref(*data_provider_)));
 
-  // Register callback to retrieve vio pipeline output from all modules.
```

Comment:
@ToniRV , where do you replace this functionality?

Reply:
So now this is done by the RosVisualizer, there is no need to use callbacks though, checkout the spinOnce in the RosVisualizer.